spec/unleash/constraint_spec.rb

```diff
@@ -61,16 +61,358 @@
 
       constraint = Unleash::Constraint.new('user_id', 'NOT_IN', ['123', '456'])
-      expect(constraint.matches_context?(context)).to be_falsey
+      expect(constraint.matches_context?(context)).to be false
 
       constraint = Unleash::Constraint.new('user_id', 'NOT_IN', ['456', '789'])
-      expect(constraint.matches_context?(context)).to be_truthy
+      expect(constraint.matches_context?(context)).to be true
+    end
+
+    it 'matches based on property STR_STARTS_WITH value' do
+      context_params = {
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'STR_STARTS_WITH', ['dev'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_STARTS_WITH', ['development'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_STARTS_WITH', ['ment'])
+      expect(constraint.matches_context?(context)).to be false
+    end
+
+    it 'matches based on property STR_ENDS_WITH value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['ment'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['development'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['dev'])
+      expect(constraint.matches_context?(context)).to be false
+    end
+
+    it 'matches based on property STR_CONTAINS value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['ment'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['dev'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['development'])
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['DEVELOPMENT'])
+      expect(constraint.matches_context?(context)).to be false
+    end
+
+    it 'matches based on property NUM_EQ value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '0.3'
```

Comment:
Bikeshedding: 

Maybe use a different var name than `env` ?
```suggestion
          distance: '0.3'
```